pkg/client/client.go

```
@@ -243,6 +246,70 @@ func ReadClouds(authOpts *AuthOpts) error {
 	return nil
 }
 
+// LoggingTransport is a custom transport that logs requests and responses
+type LoggingTransport struct {
+	Transport http.RoundTripper
+	Region    string
+	Cluster   string
+}
+
+// RoundTrip implements the http.RoundTripper interface
+func (t *LoggingTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Add X-Request-ID header
+	reqID := uuid.New().String()
+	req.Header.Set("X-Request-ID", reqID)
+
+	token := os.Getenv("API_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("API token unavailable")
+	}
+	req.Header.Set("X-Cluster", os.Getenv("CLUSTER_NAME"))
```

do we need throw an error if CLUSTER_NAME env doesnt exist ?

---

yes, we should throw an error. I have update the code.